src/enosys.c

```diff
@@ -71,6 +71,11 @@ int fi_no_domain(struct fid_fabric *fabric, struct fi_info *info,
 {
 	return -FI_ENOSYS;
 }
+int fi_no_domain2(struct fid_fabric *fabric, struct fi_info *info,
+		struct fid_domain **dom, uint64_t flags, void *context)
+{
+	return -FI_ENOSYS;
```

Comment:
This would be different from other implementations, but I wonder if this call (and ep2) should be:
```
if (flags)
    return -FI_ENOSYS;

return fi_domain(...);
```
but with a different name?

Reply:
Good point. This can also be handled in fi_domain2() and fi_endpoint2(). Or should it be done at both places? I think it's fine to keep the current name, even though it doesn't always return enosys.

Reply:
Now that you mention it, handling this in fi_domain2() would be helpful for migrating apps, except that I would handle it differently:
```
fi_domain2(...)
{
    if (!flags)
        return fi_domain(...);

    return FI_CHECK_OP(...) blah blah
}
```
Internally, we could have something like the call above that providers can point their implementation to (where !flags returns FI_ENOSYS).

Reply:
updated.